dotCMS/src/main/java/com/dotcms/rest/api/v1/portlet/PortletResource.java

```diff
@@ -0,0 +1,154 @@
+import com.dotcms.repackage.javax.ws.rs.Produces;
+import com.dotcms.repackage.javax.ws.rs.core.Context;
+import com.dotcms.repackage.javax.ws.rs.core.MediaType;
+import com.dotcms.repackage.javax.ws.rs.core.Response;
+import com.dotcms.repackage.jersey.repackaged.com.google.common.collect.ImmutableMap;
+import com.dotcms.repackage.org.glassfish.jersey.server.JSONP;
+import com.dotcms.rest.InitDataObject;
+import com.dotcms.rest.ResponseEntityView;
+import com.dotcms.rest.WebResource;
+import com.dotcms.rest.annotation.NoCache;
+import com.dotcms.rest.exception.mapper.ExceptionMapperUtil;
+import com.dotmarketing.business.APILocator;
+import com.dotmarketing.business.ApiProvider;
+import com.dotmarketing.business.DotStateException;
+import com.dotmarketing.business.portal.DotPortlet;
+import com.dotmarketing.business.portal.PortletAPI;
+import com.dotmarketing.business.web.WebAPILocator;
+import com.dotmarketing.exception.DotDataException;
+import com.dotmarketing.portlets.languagesmanager.business.LanguageAPI;
+import com.dotmarketing.portlets.languagesmanager.model.Language;
+import com.dotmarketing.util.Logger;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableSet;
+import com.liferay.portal.language.LanguageUtil;
+import com.liferay.portal.model.Portlet;
+
+import io.vavr.control.Try;
+
+
+@Path("/v1/portlet")
+@SuppressWarnings("serial")
+public class PortletResource implements Serializable {
+
+  private static final String ROLE_ID_SEPARATOR = ",";
+
+  private final WebResource webResource;
+  private final PortletAPI portletApi;
+  private final SystemMessageEventUtil systemMessageEventUtil =
+      SystemMessageEventUtil.getInstance();
+  private final LanguageAPI langApi =
+      APILocator.getLanguageAPI();
+  
+  /**
+   * Default class constructor.
+   */
+  public PortletResource() {
+    this(new WebResource(new ApiProvider()), APILocator.getPortletAPI());
+  }
+
+  @VisibleForTesting
+  public PortletResource(WebResource webResource, PortletAPI portletApi) {
+    this.webResource = webResource;
+    this.portletApi = portletApi;
+  }
+
+  @POST
+  @Path("/custom")
+  @JSONP
+  @NoCache
+  @Consumes(MediaType.APPLICATION_JSON)
+  @Produces({MediaType.APPLICATION_JSON, "application/javascript"})
+  public final Response contentPortlet(@Context final HttpServletRequest request,
+                                 final CustomPortletForm formData) {
+    InitDataObject init = webResource.init(null, true, request, true, "roles");
+
+    Response response = null;
+
+    try {
+
+      if (!UtilMethods.isSet(formData.portletId)) {
+        return ExceptionMapperUtil.createResponse(new DotStateException("Portlet Id is required"),
+                Response.Status.BAD_REQUEST);
+      }
+      if (!UtilMethods.isSet(formData.portletName)) {
+        return ExceptionMapperUtil.createResponse(new DotStateException("Portlet Name is required"),
+                Response.Status.BAD_REQUEST);
+      }
+
```

Comment:
you may have to log the entry for support such as 
Logger.log (this, ()-> "calling content portlet ....")